### PR TITLE
Cleanup large build artifacts on CI

### LINKED ISSUE #2483: Minimize footprint of K CI jobs
Currently a K checkout on CI takes roughly 10GB. Here is a breakdown:

```
# du -h --max-depth 2
3.0M    ./java-backend/target
1.7M    ./java-backend/src
4.6M    ./java-backend
12K     ./scripts
8.0K    ./kore/project
4.0M    ./kore/target
608K    ./kore/src
4.6M    ./kore
248M    ./haskell-backend/target
1.2G    ./haskell-backend/src
1.5G    ./haskell-backend
8.0K    ./target/antrun
28K     ./target
659M    ./k-distribution/pl-tutorial
633M    ./k-distribution/target
352K    ./k-distribution/include
884M    ./k-distribution/k-tutorial
8.0K    ./k-distribution/.externalToolBuilders
576K    ./k-distribution/src
2.0G    ./k-distribution/tests
24K     ./k-distribution/test-kompiled
4.1G    ./k-distribution
4.4M    ./kframework-5.2.0/java-backend
12K     ./kframework-5.2.0/scripts
4.5M    ./kframework-5.2.0/kore
1.5G    ./kframework-5.2.0/haskell-backend
384M    ./kframework-5.2.0/k-distribution
332K    ./kframework-5.2.0/nix
64K     ./kframework-5.2.0/package
9.0M    ./kframework-5.2.0/.git
8.0K    ./kframework-5.2.0/.idea
64K     ./kframework-5.2.0/src
36K     ./kframework-5.2.0/.github
249M    ./kframework-5.2.0/llvm-backend
350M    ./kframework-5.2.0/debian
1.9M    ./kframework-5.2.0/web
8.2M    ./kframework-5.2.0/kernel
132K    ./kframework-5.2.0/ktree
2.5G    ./kframework-5.2.0
332K    ./nix
12K     ./package/arch
24K     ./package/macos
44K     ./package/debian
12K     ./package/docker
108K    ./package
52K     ./.git/hooks
4.4M    ./.git/modules
4.0K    ./.git/branches
3.2M    ./.git/refs
240K    ./.git/objects
8.0K    ./.git/info
180K    ./.git/logs
9.0M    ./.git
4.0K    ./kframework-5.2.0@tmp
8.0K    ./.idea
52K     ./src/main
64K     ./src
20K     ./.github/workflows
12K     ./.github/ISSUE_TEMPLATE
36K     ./.github
572K    ./k-exercises/.git
481M    ./k-exercises/tutorial
482M    ./k-exercises
135M    ./llvm-backend/target
97M     ./llvm-backend/src
231M    ./llvm-backend
8.0K    ./web/sitemap
768K    ./web/k-web-theme
52K     ./web/pages
36K     ./web/static_content
700K    ./web/public_content
1.9M    ./web
6.0M    ./kernel/target
2.5M    ./kernel/src
8.5M    ./kernel
112K    ./ktree/target
36K     ./ktree/src
156K    ./ktree
8.8G    .
```

The biggest space eaters are:

```
633M    ./k-distribution/target
659M    ./k-distribution/pl-tutorial
884M    ./k-distribution/k-tutorial
1.2G    ./haskell-backend/src
1.5G    ./haskell-backend
1.5G    ./kframework-5.2.0/haskell-backend
2.0G    ./k-distribution/tests
2.5G    ./kframework-5.2.0
4.1G    ./k-distribution
```

So, we can probably eliminate quite a bit of storage by clearing the `kframework-${VERSION_NUMBER}` unconditionally after builds (as a `post` step). We can also probably clear the `haskell-backend/src/...` stuff after building the haskell backend, maybe this can be made a default part of the K build. Not sure if that would hurt rebuild times though.

Another good target would be `k-distribution/tests`, just making sure to run `make clean` after running all tests.

### PR DESCRIPTION
**Pull request by gtrepta**
fixes: https://github.com/runtimeverification/k/issues/2483

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -101,6 +101,19 @@ pipeline {
...
             }
           }
+          post {
+            always {
+              steps {
+                sh '''
+                  rm -rf k-distribution/k-tutorial/1_basic/build
+                  rm -rf kframework-${env.VERSION}
+                  make --directory=k-distribution/tests/regression-new clean
+                  make --directory=k-exercises clean
```
Comment (ehildenb) [inline Jenkinsfile]: Does this also clean up stuff in `k-distribution/`?

**Comment by ehildenb** [comment]
Can you also update the `.gitmodules` file while you're at it? The haskell/llvm backends have been moved to runtimeverification/llvm-backend and runtimeverification/haskell-backend.

File: Jenkinsfile [hunk 2]
```diff
@@ -101,6 +101,17 @@ pipeline {
               }
             }
           }
+          post {
+            always {
+              sh '''
+                rm -rf k-distribution/k-tutorial/1_basic/build
+                rm -rf kframework-${env.VERSION}
```
Comment (ehildenb) [inline Jenkinsfile]:
```suggestion
                rm -rf kframework-${VERSION}
```

Inside groovy scripts, you have to use the `${NAMESPACE.VARNAME}` for substitutions. But in bash blocks, the varibales `${env.VARNAME}` are exposed as environment variables.

File: Jenkinsfile (hunk 1 above)
Comment (gtrepta) [reply Jenkinsfile]: the `make --directory=k-exercises clean` command should clean up `k-distribution/pl-tutorial`. But that's all besides the other two commands that deal with that subdirectory.

I guess `make --directory=k-distribution clean` would cover more of it. @ehildenb Should we also try to clean up `k-distribution/target`? I think that's ~500M.

**Comment by ehildenb** [comment]
Down to 1.6GB:

```
$ du -h --max-depth 1 K_PR-2487 K_PR-2487@tmp/
4.6M    K_PR-2487/kore
156K    K_PR-2487/ktree
36K     K_PR-2487/.github
1.7M    K_PR-2487/k-exercises
8.0K    K_PR-2487/.idea
415M    K_PR-2487/haskell-backend
28K     K_PR-2487/target
750M    K_PR-2487/k-distribution
112K    K_PR-2487/package
12K     K_PR-2487/scripts
8.5M    K_PR-2487/kernel
4.6M    K_PR-2487/java-backend
64K     K_PR-2487/src
251M    K_PR-2487/llvm-backend
332K    K_PR-2487/nix
4.0K    K_PR-2487/kframework-5.2.0@tmp
9.3M    K_PR-2487/.git
1.9M    K_PR-2487/web
1.6G    K_PR-2487
4.0K    K_PR-2487@tmp/
```

**Comment by ehildenb** [comment]
@gtrepta one more thing you can do is run `make clean` as the last step in the `test-in-container` script. Approving anyway, because it looks good.

**Review by ehildenb — APPROVED** [review]

**Comment by ehildenb** [comment]
Looks like that one didn't change it much, but still, 1.6GB is much better.

```
$ du -h --max-depth 1 K_PR-2487* | sort -h
4.0K    K_PR-2487/kframework-5.2.0@tmp
4.0K    K_PR-2487@tmp
8.0K    K_PR-2487/.idea
12K     K_PR-2487/scripts
28K     K_PR-2487/target
36K     K_PR-2487/.github
64K     K_PR-2487/src
112K    K_PR-2487/package
156K    K_PR-2487/ktree
332K    K_PR-2487/nix
1.7M    K_PR-2487/k-exercises
1.9M    K_PR-2487/web
4.6M    K_PR-2487/java-backend
4.6M    K_PR-2487/kore
8.5M    K_PR-2487/kernel
9.3M    K_PR-2487/.git
251M    K_PR-2487/llvm-backend
415M    K_PR-2487/haskell-backend
750M    K_PR-2487/k-distribution
1.6G    K_PR-2487
```

**Pull request merged**